locket/__init__.py

```diff
@@ -150,19 +150,32 @@ def __init__(self, path):
         self._file = None
 
     def acquire(self, timeout=None, retry_period=None):
-        if self._file is None:
-            self._file = open(self._path, "wb")
-        if timeout is None and _lock_file_blocking_available:
-            _lock_file_blocking(self._file)
+        new_fd = self._file is None
+        fd = open(self._path, "wb")  if new_fd  else self._file
```

Comment:
I find this really awkward, but we do not want to close `self._file` if we have fail to acquire our own lock. Definitely open to suggestions on an alternative implementation.